rotkehlchen/chain/ethereum/utils.py

```diff
@@ -208,11 +212,25 @@ def try_download_ens_avatar(
     if avatar_url == '':
         return  # Avatar is not set
 
-    try:
-        avatar = requests.get(avatar_url, timeout=DEFAULT_TIMEOUT_TUPLE).content
-    except RequestException as e:
-        log.error(f'Got error {str(e)} during querying ens avatar for {ens_name}.')
-        return
+    avatar = None
+    if avatar_url.startswith('eip155'):  # an NFT is set. Let's try first ENS app's own metadata
+        try:
+            avatar = requests.get(f'{ENS_METADATA_URL}/avatar/{ens_name}', timeout=DEFAULT_TIMEOUT_TUPLE).content  # noqa: E501
+        except RequestException as e:  # Try opensea -- if we got it
+            log.error(f'Got error {str(e)} during querying ENS app for NFT avatar for {ens_name}. May fall back to opensea')  # noqa: E501
+            if opensea is None:
+                return  # no opensea
+            avatar_url = opensea.get_nft_image(avatar_url)
+            if avatar_url is None:
+                return  # no luck with opensea
+            # proceed to query the new avatar url
+
+    if avatar is None:  # we have not populated it via an NFT query above yet
+        try:
+            avatar = requests.get(avatar_url, timeout=DEFAULT_TIMEOUT_TUPLE).content
+        except RequestException as e:
+            log.error(f'Got error {str(e)} during querying ens avatar {avatar_url} for {ens_name}.')  # noqa: E501
```

Comment:
question, is during querying correct?

```suggestion
            log.error(f'Got error {str(e)} while querying ens avatar {avatar_url} for {ens_name}.')  # noqa: E501
```

Reply:
yes it's correct

Reply:
But to be honest while does sound better. I think during woudl be more matched to something like: "During the query